### PR TITLE
Search documents

### PR DESCRIPTION
**Pull request by ovidiu-ostafi**

### CODE REVIEW

**Comment by Frozire** [comment]
Looking good, @ovidiu-ostafi .

~~I assume we are storing the Parser Document class "data" as a property, due to some limitation, or already set way, of WordPress? It seems like quite a hassle when saving and getting values by looping through it's fields?~~

Edit: Just found this comment 

> "The 'matches' key contains an array of objects that have the 'fields' value accessible directly through the magic method __get(). This means that you can call any field from the 'fields' with the direct pointer:"

Everything is fine then. I guess there is no reason to merge this yet, but so far it's looking good.

**Comment by ovidiu-ostafi** [comment]
@Frozire I have double checked this and I think we can spare some loops as you suggested.

The main results are like this: https://github.com/BenjaminMedia/wp-cxense/tree/search_documents#widget-documents (look at the returning format). If we can skip (do not need) the id, siteId and score from the results and just use the fields array then the main looping method https://github.com/BenjaminMedia/wp-cxense/blob/search_documents/src/Bonnier/WP/Cxense/Parsers/Document.php#L29 (the one that loops at each document through the above mentioned fields) can be transformed into just a simple assignment of 'fields':

``` php
       private function assignData(\stdClass $objData) {
		$this->fields = $objData->fields;
	}
```

This way the magic method __get() will still return the cxense required fields (from the call) and we don't loop at all.

The main problem is that I am not sure if they need these fields (id, siteId, score) on the frontend.

Another approach I guess would be to just skip the Document parser all together and return the exact response from the cxense, sadly this is a really bad situation since our cxense fields looks like:

``` php
public 'fields' => 
    array (size=9)
      0 => 
        object(stdClass)[558]
          public 'field' => string 'author' (length=6)
          public 'value' => string 'i form' (length=6)
      1 => 
        object(stdClass)[561]
          public 'field' => string 'bod-cat' (length=7)
          public 'value' => string 'Sund graviditet' (length=15)
```

Imagine the loops that we need to do in frontend just to get a field call 'author', to me this is not a valid solution.

**Comment by Frozire** [comment]
@ovidiu-ostafi 

Right, your current solution is the most desirable then, definitely want to keep that logic away from the frontend.

As for the id, siteid and score, I think we can keep this logic abstracted away too. If we want to sort by scoring, we would want to do this sorting in the backend, ahead of handing it to the frontend. The siteid isn't needed since the site knows itself, and the id would only be for logging.

So yeah, the frontend won't be needing those variables, so changing it to a simple assignment of fields is possible. But I will leave that assessment up to you; I will gladly accept both solutions.

**Review by Frozire — DISMISSED** [review]

File: src/Bonnier/WP/Cxense/Services/DocumentSearch.php [hunk 1]
```diff
@@ -0,0 +1,325 @@
...
+	 */
+	private $objSettings;
+	
+	/**
+	 * Returning fields
+	 *
+	 * @var array $arrFields
+	 */
+	private $arrFields = [
+		'bod-cat',
```
Comment (alfhen) [inline src/Bonnier/WP/Cxense/Services/DocumentSearch.php]: I noticed that you have ```bod``` hardcoded in the ```bod-cat``` value and the other fields.
The problem with this is that ```bod``` is a company specific prefix, as this plugin has to be used by several companies the prefix might be different for each company. Which is why the plugin contains a setting called ```organisation_prefix``` and this is used to prefix the various values.
Therefore we need to figure out a way to set these fields with the prefix gotten from the settings page.
``` 
'bod-cat',
'bod-cat-top',
'bod-pagetype',
'bod-cat-url',
```

**Review by alfhen — COMMENTED** [review]
Looks good 😄  I only have the one comment that you will see.

File: src/Bonnier/WP/Cxense/Services/DocumentSearch.php (hunk 1 above)
Comment (ovidiu-ostafi) [reply src/Bonnier/WP/Cxense/Services/DocumentSearch.php]: Can you check this one out: https://github.com/BenjaminMedia/wp-cxense/pull/7/commits/53c2343bb92bca7a0535e0f510c4f189ebcb8882 ?

**Review by alfhen — APPROVED** [review]

**Pull request merged**